Source/loadsave.cpp

```diff
@@ -682,43 +682,41 @@ void LoadMissile(LoadHelper *file, Missile &missile)
 	missile.limitReached = file->NextBool32();
 }
 
-void LoadObject(LoadHelper *file, int i)
+void LoadObject(LoadHelper &file, Object &object)
 {
-	Object *pObject = &Objects[i];
-
-	pObject->_otype = static_cast<_object_id>(file->NextLE<int32_t>());
-	pObject->position.x = file->NextLE<int32_t>();
-	pObject->position.y = file->NextLE<int32_t>();
-	pObject->_oLight = file->NextBool32();
-	pObject->_oAnimFlag = file->NextLE<uint32_t>();
-	file->Skip(4); // Skip pointer _oAnimData
-	pObject->_oAnimDelay = file->NextLE<int32_t>();
-	pObject->_oAnimCnt = file->NextLE<int32_t>();
-	pObject->_oAnimLen = file->NextLE<uint32_t>();
-	pObject->_oAnimFrame = file->NextLE<uint32_t>();
-	pObject->_oAnimWidth = file->NextLE<int32_t>();
-	file->Skip(4); // Skip _oAnimWidth2
-	pObject->_oDelFlag = file->NextBool32();
-	pObject->_oBreak = file->NextLE<int8_t>();
-	file->Skip(3); // Alignment
-	pObject->_oSolidFlag = file->NextBool32();
-	pObject->_oMissFlag = file->NextBool32();
-
-	pObject->_oSelFlag = file->NextLE<int8_t>();
-	file->Skip(3); // Alignment
-	pObject->_oPreFlag = file->NextBool32();
-	pObject->_oTrapFlag = file->NextBool32();
-	pObject->_oDoorFlag = file->NextBool32();
-	pObject->_olid = file->NextLE<int32_t>();
-	pObject->_oRndSeed = file->NextLE<uint32_t>();
-	pObject->_oVar1 = file->NextLE<int32_t>();
-	pObject->_oVar2 = file->NextLE<int32_t>();
-	pObject->_oVar3 = file->NextLE<int32_t>();
-	pObject->_oVar4 = file->NextLE<int32_t>();
-	pObject->_oVar5 = file->NextLE<int32_t>();
-	pObject->_oVar6 = file->NextLE<uint32_t>();
-	pObject->bookMessage = static_cast<_speech_id>(file->NextLE<int32_t>());
-	pObject->_oVar8 = file->NextLE<int32_t>();

```

Comment:
The diff from github is absolutely terrible here. I used VS to compare:
![image](https://user-images.githubusercontent.com/461579/132143001-b7eb2a43-6c2f-4c9b-ab8e-1aa2f17ae0d8.png)

Reply:
I use Diffuse